src/test/ui/consts/const-mut-refs/mut_ref_in_final.rs

```diff
@@ -0,0 +1,24 @@
+#![feature(const_mut_refs)]
+#![feature(const_fn)]
+#![feature(raw_ref_op)]
+const NULL: *mut i32 = std::ptr::null_mut();
+const A: *const i32 = &4;
+
+// It could be made sound to allow it to compile,
+// but we do not want to allow this to compile,
+// as that would be an enormous footgun in oli-obk's opinion.
+const B: *mut i32 = &mut 4; //~ ERROR untyped pointers are not allowed in constant
+
+// Could be ok, but the same analysis that prevents the mutable one above will also bail out here
+// Using a block with some complex content, because just `&45` would get promoted,
+// which is not what we want to test here.
+const C: *const i32 = &{ //~ ERROR untyped pointers are not allowed in constant
+    let mut x = 42;
+    x += 3;
+    x
+};
+
```

Comment:
Could you add tests for what you said in the PR description, "Some(&mut 42) is not ok, while None::<&mut i32> is"?

Reply:
This comment seems to still be open?